docs/packaging: Update release process for package (un)deprecation (#549)
getsolus/help-center-docs: docs/packaging/procedures/release-processes.md

Follow these steps to submit a package for deprecation:

> 1. Create a new branch on the monorepo [on GitHub](https://github.com/getsolus/packages) with a suitable name e.g. deprecate-foobar

Reviewer: ```suggestion
1. Create a new branch in the monorepo [on GitHub](https://github.com/getsolus/packages) with a suitable name, for example deprecate-foobar
```
I think the guideline is to use no Latin abbreviations like "e.g." and "i.e."

Should there be a note to fork it, for external contributors?

> 1. Create a new branch in the monorepo [on GitHub](https://github.com/getsolus/packages) (forking the repository first if necessary) with a suitable name such as deprecate-foobar

Reviewer: ```suggestion
1. Create a new branch in the Solus `packages` repository [on GitHub](https://github.com/getsolus/packages) (forking it first if necessary) with a suitable name such as "deprecate-foobar"
```

I remembered that "monorepo" is also a no-no

2. Edit `repo_data/distribution.xml.in`
3. Add the package you wish to deprecate at the end of the `<Obsoletes>` section
4. Change directory to repo_data (`cd repo_data`), then run the `merged_repos.sh` script

> 5. Remove the package from the monorepo e.g. `rm -fr packages/f/foobar`

Reviewer: ```suggestion
5. Remove the package from the monorepo, for example `rm -fr packages/f/foobar`
```

> 5. Remove the package from the monorepo, for example `rm -fr packages/f/foobar`

Reviewer: ```suggestion
5. Remove the package from the Solus `packages` repository, for example `rm -fr packages/f/foobar`
```

5. Commit and push your changes to your repository/branch
6. Open a Pull Request on GitHub

### Un-deprecating a package

The above steps also apply to un-deprecating a package, with the following additions:

> 1. Create a new branch on the monorepo [on GitHub](https://github.com/getsolus/packages) with a suitable name e.g. undeprecate-foobar

Reviewer: ```suggestion
1. Create a new branch in the monorepo [on GitHub](https://github.com/getsolus/packages) with a suitable name, for example undeprecate-foobar
```

> 1. Create a new branch in the monorepo [on GitHub](https://github.com/getsolus/packages) (forking the repository first if necessary) with a suitable name such as undeprecate-foobar

Reviewer: ```suggestion
1. Create a new branch in the Solus `packages` repository [on GitHub](https://github.com/getsolus/packages) (forking it first if necessary) with a suitable name such as "undeprecate-foobar"
```

> 2. Find the commit deprecating the package and restore the package within the monorepo structure. e.g. Restore all files in packages/f/foobar or use `git revert` if appropriate

Reviewer: ```suggestion
2. Find the commit deprecating the package and restore the package within the monorepo structure. For example restore all files in packages/f/foobar or use `git revert` if appropriate
```

> 2. Find the commit deprecating the package and restore the package within the monorepo structure. For example restore all files in packages/f/foobar or use `git revert` if appropriate

Reviewer: ```suggestion
2. Find the commit deprecating the package and restore the package within the repository structure. For example restore all files in packages/f/foobar or use `git revert` if appropriate
```

3. Edit `repo_data/distribution.xml.in` and remove the package you wish to undeprecate in the `<Obsoletes>` section
4. Change directory to repo_data (`cd repo_data`), then run the `merged_repos.sh` script

> 5. Open a Pull Request on GitHub and ask a Solus Staff member to use available tooling to update the repository files, then index the repo with `ferryctl index unstable` (or do it yourself if you have the ability)

Reviewer: ```suggestion
5. Open a Pull Request on GitHub and ask a Solus Staff member to use available tooling to update the repository files, then index the repository with `ferryctl index unstable` (or do it yourself if you have the ability)
```
